includes/class-edd-logging.php

```diff
@@ -151,13 +151,15 @@ public function insert_log( $log_data = array(), $log_meta = array() ) {
 		// Used to dynamically dispatch the method call to insert() to the correct class.
 		$insert_method = 'edd_add_log';
 
-		// Set up variables to hold data to go into the logs table by default
+		// Set up variables to hold data to go into the logs table by default.
 		$data = array(
-			'message'     => $args['post_content'],
-			'object_id'   => $args['post_parent'],
-			'object_type' => isset( $args['object_type'] )
-				? $args['object_type']
-				: 'download',
+			'content'     => $args['post_content'],
+			'object_id'   => isset( $args['post_parent'] )
+				? $args['post_parent']
+				: 0,
+			'object_type' => isset( $args['log_type'] )
+				? $args['log_type']
+				: 'unknown',
```

Comment:
I think I'd rather prefer a `null` or something, that way we can filter later and do queries against things as `NULL` or `NOT NULL`.

Reply:
I updated that, but just wanted to mention that it needs to be changed to `null` in several places, as 0 was the standard for "none" previously to EDD 3.0. See changes here:

https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7284/commits/9b08893c31d692de68a0e582936f77e499dfa710

Reply:
@mintplugins I'm fine with the object ID being 0, I was referencing the `object_type`. I'd like that to be `null` instead of `unknown`.

Reply:
Ah I see. One of the problems there is that a type is required. So we can't pass null, unless we decide that a type is not required. 

See type being required here:
https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7284/files#diff-22827a25e9cea91ff2d49d62f066894eR51

Reply:
(PS I [reverted](https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7284/commits/ff5c1a8b47dddfdfa0f6da00a02358a053d4aa51) https://github.com/easydigitaldownloads/easy-digital-downloads/commit/9b08893c31d692de68a0e582936f77e499dfa710

Reply:
@mintplugins I'm fine with updating that to be allowed to be `null` only, so maybe we just do a check for empty and not is_null. Also, we should only allow a null type so long as the object ID is `0`.

Reply:
@cklosowski See those changes as requested here: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/7284/commits/f2250700ab971d68f8e9db560dfdc06aa3a3202a